ateam_kenobi/src/kenobi_node.cpp

```diff
@@ -0,0 +1,266 @@
+// Copyright 2021 A Team
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+
+#include <tf2/convert.h>
+#include <tf2/utils.h>
+#include <chrono>
+#include <functional>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_components/register_node_macro.hpp>
+#include <ateam_msgs/msg/ball_state.hpp>
+#include <ateam_msgs/msg/robot_state.hpp>
+#include <ateam_msgs/msg/field_info.hpp>
+#include <ateam_msgs/msg/robot_motion_command.hpp>
+#include <ateam_msgs/msg/overlay.hpp>
+#include <ateam_msgs/msg/world.hpp>
+#include <ateam_common/game_controller_listener.hpp>
+#include <ateam_common/topic_names.hpp>
+#include <ateam_common/indexed_topic_helpers.hpp>
+#include <ateam_geometry/types.hpp>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+#include "types/world.hpp"
+#include "types/message_conversions.hpp"
+#include "play_selector.hpp"
+#include "visualization/overlay_publisher.hpp"
+#include "in_play_eval.hpp"
+#include "motion/world_to_body_vel.hpp"
+
+namespace ateam_kenobi
+{
+
+using ateam_common::indexed_topic_helpers::create_indexed_publishers;
+using ateam_common::indexed_topic_helpers::create_indexed_subscribers;
+using namespace std::literals::chrono_literals;
+
+class KenobiNode : public rclcpp::Node
+{
+public:
+  explicit KenobiNode(const rclcpp::NodeOptions & options = rclcpp::NodeOptions())
+  : rclcpp::Node("kenobi_node", options),
+    overlay_publisher_("Kenobi", *this),
+    play_info_publisher_(*this),
+    play_selector_(overlay_publisher_, play_info_publisher_),
+    game_controller_listener_(*this)
+  {
+    declare_parameter<bool>("use_world_velocities", false);
+
+    create_indexed_subscribers<ateam_msgs::msg::RobotState>(
+      blue_robots_subscriptions_,
+      Topics::kBlueTeamRobotPrefix,
+      10,
```

Comment:
A bunch of these subscribers use the same value for the QOS, do we want to make this a constant/variable that they share?

Reply:
meh 🤷 we can, but it's also not that bad to not